fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2490,6 +2493,9 @@ public LoanOverdueDTO applyChargeToOverdueLoanInstallment(final Long loanId, fin
 
                 final LoanCharge loanCharge = LoanCharge.createNewFromJson(loan, chargeDefinition, command, entry.getValue());
 
+                if(BigDecimal.ZERO.compareTo(loanCharge.amount()) >= 0){
```

Comment:
This would be also true not only if `loanCharge.amount()` is 0 but also if it's negative. Is that really intended?

Reply:
Not really. I did it just to be safe.

Reply:
would it be clearer for future readers of this code to check only for `equals(ZERO)` instead? Just a thought.

Reply:
Sure. I will make the change

Reply:
`equals` won't work because it also takes into consideration the scale. I have changed it to use `compareTo == 0`

Reply:
OK, cool - resolves conversation for me; thanks!